src/engine.h

```diff
@@ -162,6 +170,11 @@ class Engine {
                                    const uint32_t binding,
                                    ResourceInfo* info) = 0;
 
+  /// Copy the contents of all resources bound to descriptors and get
+  /// their resource information e.g., size for buffer, width, height,
+  /// depth for image.
+  virtual std::vector<ResourceInfo> GetAllDescriptorInfo() = 0;
```

Comment:
I think we want to get just a single buffer. If we have one small one to extract and a huge second one there is no reason to extract the huge buffer.